File: typescript/src/structural/facade/publisher/Storage.ts
import { Article } from "../interfaces/IArticle";

function sleep(ms: number): Promise<void> {
    return new Promise(resolve => setTimeout(resolve, ms));
}

async function saveToDb(article: Article): Promise<void> {
    console.log(`Uploading to the database: ${article.slug}`);
    await sleep(3000);
    console.log("Completed");
}

function publishToCdn(article: Article): void {
    console.log(`Publishing in CDN: ${article.slug} with HTML content`);
    console.log(`${article.html}`);
}

export { saveToDb, publishToCdn, Article };

File: python/structural/facade/publisher/storage.py
import time

def save_to_db(article: dict):
    print(f"Uploading to the database: {article['slug']}")
    time.sleep(3)
    print("Completed")

def publish_to_cdn(article: dict):
    print(f"Publishing in CDN: {article['slug']} with HTML content")
    print(f"{article['html']}")
